packages/l/libcurl/xmake.lua

```diff
@@ -30,11 +31,14 @@ package("libcurl")
     add_configs("libssh2",  {description = "Use libSSH2 library.", default = false, type = "boolean"})
     add_configs("libpsl",   {description = "Use libpsl library.", default = false, type = "boolean"})
 
+    if is_plat("android") and is_host("windows") then
+        add_deps("ninja")
+        set_policy("package.cmake_generator.ninja", true)
+    end
+
     -- we init all configurations in on_load, because package("curl") need it.
     on_load(function (package)
-        if package:is_plat("linux", "android", "cross") then
-            package:config_set("openssl", true)
-        end
```

Comment:
Removing those lines broke existing xmake projects doing `add_requires("libcurl")`

Reply:
It works fine on my device (linux, x86_64), can you provide more information?

Reply:
`add_requires("libcurl")` on Linux no longer links with libcurl and no longer has HTTPS support, resulting in `unsupported protocol` errors.

before your change, openssl support was enabled by default

Reply:
This does break the default HTTPS support, but this is expected because
1. HTTPS support should not be forced, curl builds do not enforce ssl
2. There are other options now (openssl3)

Downstreams should choose whether to enable encryption, and which one to enable.

Reply:
If one must be turned on by default, it is difficult to choose which one to turn on. I think it is best to let the user choose.

Reply:
I understand it's expected, but it's problematic to update a package and break the existing behavior, especially here where all xmake projects using libcurl will get a failure on the broadly used https protocol.

openssl should still be the default when no option is set, I think we can detect the case where openssl and openssl3 are not set (nil) and enable openssl as before for retrocompatibility.

@waruqi is this possible?

Reply:
I think the ability to build libcurl without SSL support needs to be preserved, is there a way for xmake to tell if a certain config is set "by default"? Otherwise I think it's a good idea to automatically enable openssl.

Reply:
I think so but I need confirmation from xmake author, afaik configs can have three values : true, false or nil (not set).
what I'm suggesting is basically:
```lua
if package:config("openssl") == nil and package:config("openssl3") == nil then
    package:config_set("openssl", true)
end
```

this will support `add_requires("libcurl", { openssl = false })` as well

Reply:
If openssl is not explicitly set and no default value is configured in add_configs, it defaults to nil